ops/main.py

```diff
@@ -42,6 +43,114 @@
 logger = logging.getLogger()
 
 
+@dataclasses.dataclass(frozen=True)
+class _JujuContext:
+    # source: https://juju.is/docs/juju/charm-environment-variables
+    # the HookVars function: https://github.com/juju/juju/blob/3.6/worker/uniter/runner/context/context.go#L1398
+    # only allow accesing these vars because these are the currently used vars in main.py
+
+    # env var JUJU_ACTION_NAME
+    action_name: Optional[str] = None
+
+    # env var JUJU_ACTION_UUID
+    action_uuid: Optional[str] = None
+
+    # env var JUJU_CHARM_DIR
+    charm_dir: Optional[str] = None
+
+    # env var JUJU_DEBUG
+    debug: Optional[str] = None
+
+    # env var JUJU_DEPARTING_UNIT
+    departing_unit: Optional[str] = None
+
+    # env var JUJU_DISPATCH_PATH
+    dispatch_path: Optional[str] = None
```

Comment:
It's added [here](https://github.com/juju/juju/blame/8ad001fadb526945445506026aea5c828f9320d7/worker/uniter/runner/runner.go#L471) and from 2.7.4+ on it can't be None.